### DISCUSSION TITLE
(IDF 5.0.2) ESP32S2 apa106.py compile error

### DISCUSSION THREAD

**Discussion by fcolmant** [micropython/micropython#13151] [ESP32]
Hello,

I'm working on macOS 14.0 / IDF 5.0.2. 

When I do compile v21 of Micropython for ESP32S2 (or for ESP32), I do get this error :
```
MPY apa106.py
error compiling apa106.py:

ninja: build stopped: subcommand failed.
```
If I get rid of the apa106.py file, the compilation gets stuck at :
```
MPY flashbdev.py
error compiling flashbdev.py:

ninja: build stopped: subcommand failed.
```

The specific steps I do take are :
```
git clone https://github.com/_user-name_/micropython
cd micropython
git remote add upstream https://github.com/micropython/micropython
git checkout -b dev-branch
cd mpy-cross
make
cd ../ports/esp32 
make BOARD=ESP32_GENERIC_S2 submodules
make BOARD=ESP32_GENERIC_S2
```

It's the second time I try to compile Micropython. The first time was a year or 2 ago, and it went flawlessly. This time I really don't know what to do.

Any help would be appreciated.

Cheers,

François.

**Comment by sophiedegran** [comment]
This file 'only' import neopixel.. and is inside ports/esp32/modules
Perharps do a 'make clean' before
if not there is pb in import neopixel

**Comment by MATTYGILO** [comment]
Did you find a solution to this?

**Comment by robert-hh** [comment]
It may be that mpy-cross is missing. Did you build it before?

**Comment by Josverl** [comment]
I suggest to give [mpbuild ](https://github.com/mattytrentini/mpbuild) a try as that removes the hassle of setting up the toolchain and the correct IDFs.
you need to have docker (or similar) running, and the MicroPython repo cloned;
then from the repo run: 
```
mpbuild clean ESP32_GENERIC_S2 # to clear out old partial builds 
mpbuild build ESP32_GENERIC_S2
``` 

If you need a specific IDF version , then you can try that after you have your first successful 'default' build

**Comment by MATTYGILO** [comment]
@robert-hh I have mpy-cross built, @Josverl I am using a custom board so I can't use a generic one

Here is my build output:
[output.txt](https://github.com/user-attachments/files/20711130/output.txt)

This is the repo that is being built:
[https://github.com/MATTYGILO/tflite-micropython-esp](https://github.com/MATTYGILO/tflite-micropython-esp)


Note I am building this on Macbook Pro M1
